darshan-runtime/doc/darshan-runtime.txt

```diff
@@ -478,6 +446,147 @@ git submodule update --init
+
+* DARSHAN_APP_EXCLUDE: specifies a list of comma-separated regexes that match application names that should not be instrumented
+* DARSHAN_APP_INCLUDE: specifies a list of comma-separated regexes that match application names that should be instrumented (i.e., overrides DARSHAN_APP_EXCLUDE regexes)
+
+For example, the following settings would disable the generation of Darshan logs for `git` and `ls`.
+
+----
+export DARSHAN_APP_EXCLUDE="git,ls"
+----
+
+Darshan also allows for instrumentation to be disabled on specific application ranks.
+Users may set an exclusion list as well as an overriding inclusion list. Independent
+ranks or rank ranges may be specified, with ranges in the formatted like "start:end"
+(if start or end are not specified, the first or last rank is assumed, respectively).
+Note that the Darshan library will still run on all processes of an application,
+this setting just controls whether specific ranks are storing instrumentation data.
+
+* DARSHAN_RANK_EXCLUDE: specifies a list of comma-separated rank specifiers which should not be instrumented
+* DARSHAN_RANK_INCLUDE: specifies a list of comma-separated rank specifiers which should be instrumented (i.e., overrides DARSHAN_RANK_EXCLUDE specifiers)
+
+For example, the following settings would disable Darshan instrumentation for all
+application ranks except for 0-9:
+
+----
+export DARSHAN_RANK_EXCLUDE="0:"  # disable for all ranks first
+export DARSHAN_RANK_INCLUDE="0:9" # selectively enable a rank range
+----
+
+Users can also configure a couple of filters for DXT trace data which
+control whether trace records are kept or discarded at application shutdown
+time. This is useful for focusing DXT instrumentation data on file access
+patterns that are of particular interest from an I/O performance perspective:
+
+* DARSHAN_DXT_SMALL_IO_TRIGGER: specifies a floating point percentage (i.e., ".8" would be 80%) of small I/O operations (defined as operations smaller than 10 KiB). Traces are only retained for files that exhibit a percentage of small I/O operations greater than this threshold.
+* DARSHAN_DXT_UNALIGNED_IO_TRIGGER: specifies a floating point percentage (i.e., ".8" would be 80%) of unaligned I/O operations (not aligned to file alignment value determined by Darshan). Traces are only retained for files that exhibit a percentage of unaligned I/O operations greater than this threshold.
+
+For example, the following settings would ensure traces are only kept for files
+that were accessed using small I/O operations 20+% of the time:
+
+----
+export DARSHAN_DXT_SMALL_IO_TRIGGER=".2"
+----
+
+=== Darshan library config file
+
+Darshan also provides the ability for users to specify a configuration file to
+load library settings from. It is important to note that for settings that are
+specified in both the environment and via a config file, the environment
+settings will take precedence.
+
+Users may specify the config file location using the `DARSHAN_CONFIG_PATH`
```

Comment:
Just in case you missed this in my back-to-back-to-back comments from above: https://github.com/darshan-hpc/darshan/pull/703#issuecomment-1088074029

I opted for not providing a default config file installed with Darshan for the time being just because I wasn't sure the additional complexity was worth the effort, but we can revisit if I'm missing an obvious benefit. If we added a default config file, it would really just be duplicating hard-coded settings in the code base, so not sure what the benefits are for that additional wrinkle beyond just providing an example users could build off of?

Reply:
Whoops yeah, I missed it.  Let's stick with that.  In the documentation should we maybe suggest (for end users) that if they want to provide their own config file, that they check to see if the DARSHAN_CONFIG_PATH env var is already in use by the facility (like if the ALCF provided one in their site darshan module), and if so copy that one to use as a starting point?  Just a safety check to make sure that a user doesn't accidentally alter unrelated things by replacing a site config.

Reply:
I added a note about this to the docs in 3442f42. I think the NOTE could be in a better spot, but I'm still planning another pass over the docs and will fix that up as part of your other feedback.